### PR TITLE
Use _Backend from matplotlib to simplify the code

### PR DESCRIPTION
**Pull request by ianhi**
Follows how the nbagg backend currently handles things: https://github.com/matplotlib/matplotlib/blob/04f78b84a3fda987b4ac65da296d7e4aa764d894/lib/matplotlib/backends/backend_nbagg.py#L225

Conversation with @tacaswell here https://gitter.im/matplotlib/matplotlib?at=603c83fc44f5a454a447c6e9 left me with the impression that it is ok to use the private class.

This ended up being most just updating the code to match nbagg, but my taking the new show function this also fixes: https://github.com/matplotlib/ipympl/issues/303

I also created UAT.ipynb for user automated tests. This is helpful for things like the issue in #303 that are important to test as they may break again but don't really make sense as an example.

### CODE REVIEW

**Comment by ianhi** [comment]
Perhaps using `_Backend` is not optimal see: https://github.com/matplotlib/matplotlib/issues/19605#issuecomment-787919282

Though I still think it is currently the best solution

**Comment by martinRenou** [comment]
Do you think we should put `UAT.ipynb` under a test directory?

**Comment by ianhi** [comment]
> Do you think we should put UAT.ipynb under a test directory?

I ceated a top level `test-notebooks` directory because it doesn't feel like it would belong in a proper pytest `tests` directory, but happy to change.

I also moved the pytest settings into pyproject.toml to consolidate files.

**Comment by ianhi** [comment]
Bump of this @martinRenou and also @tacaswell in case this is something that ipympl really should **not** be doing?

I think having a UAT notebook is very a nice thing, but we can also swap that into a new PR if need be.

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,6 @@ js/*.tgz
 
 # OS X
 .DS_Store
+
+# vscode
+.vscode/
```
Comment (tacaswell) [inline .gitignore:18]:
```suggestion
.vscode/

```

**Review by tacaswell — APPROVED** [review]
modulo my cranky end with a new line suggestion

**Comment by ianhi** [comment]
I think the LGTM has stalled out, and it was green before adding the newline so going for it

**Pull request merged**